service/history/timerQueueTaskExecutorBase.go

```diff
@@ -76,11 +76,12 @@ func newTimerQueueTaskExecutorBase(
 }
 
 func (t *timerQueueTaskExecutorBase) executeDeleteHistoryEventTask(
+	pctx context.Context,
 	task *persistencespb.TimerTaskInfo,
 ) (retError error) {
-
 	ctx, cancel := context.WithTimeout(context.Background(), taskTimeout)
```

Comment:
pctx?

Reply:
and maybe instead of 
```go
pctx context.Context,
...
ctx, cancel := context.WithTimeout(pctx, taskTimeout)
```
just do 
```
ctx context.Context,
...
ctx, cancel := context.WithTimeout(ctx, taskTimeout)
```